docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,106 @@
+## CTCLoss <a name="CTCLoss"></a>
+
+**Versioned name**: *CTCLoss-4*
+
+**Category**: Sequence processing
+
+**Short description**: *CTCLoss* computes the CTC (Connectionist Temporal Classification) Loss.
+
+**Detailed description**:
+
+This operation is similar [Reference](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/ctc_loss)
+
+*CTCLoss* calculates loss between input and target sequences. It sums over the probability of possible alignments of input to target.
+
+Input sequences `inputs[:,i,:]` can have different length. The lengths of sequences are coded as values 1 and 0 in the second input tensor `sequence_mask`. 
+Value `sequence_mask[j, i]` specifies whether there is a sequence symbol at index `j` in the sequence `i` in the batch of sequences. 
+If there is no symbol at `j`-th position `sequence_mask[j, i] = 0`, and `sequence_mask[j, i] = 1` otherwise. 
+Starting from `j = 0`, `sequence_mass[j, i]` are equal to 1 up to the particular index `j = last_sequence_symbol`, 
+which is defined independently for each sequence `i`. For `j > last_sequence_symbol`, values in `sequence_mask[j, i]` are all zeros.
+A lenght of target sequence from `labels[i,:]` must not be greater than a lenght of corresponding input sequence. Otherwise, the operation behaves unexpectedly.
+
+*CTCLoss* calculation scheme:
+
+1. Compute probability of `j`-th character at time step `t` for `i`-th input sequence from `inputs` using softmax formula:
+\f[
+p_{t,i,j} = \frac{\exp(inputs[t,i,j])}{\sum^{K}_{k=0}{\exp(inputs[t,i,k])}}
+\f]
+
+2. For a given `i`-th target from `labels[i,:]` find all aligned paths.
+A path `S = (c1,c2,...,cT)` is aligned with a target `G=(g1,g2,...,gT)` if both chains are equal after decoding.
+The decoding removes pads `-1` from a target `G` and merges repeated characters in `G` in case *preprocess_collapse_repeated* equal to True.
```

Comment:
Why the value `-1` is a pad? Do you mean a blank character here?

Reply:
That is a format of predicted target we got from CTCGreedyDecoder. That is a way how to mark an end of target sequence. Blank character and `-1` are different terms.

Reply:
I double checked in mkldnn sources that `labels` from `CTCGreedyDecoder` does not contain `blank` index and pads with `-1`.